plugins/kubernetes/app/models/kubernetes/resource.rb

```diff
@@ -436,12 +436,19 @@ def request_delete
         delete_pods { super }
       end
 
-      # FYI per docs it is supposed to use batch api, but extension api works
       def client
         @deploy_group.kubernetes_cluster.client('batch/v1')
       end
     end
 
+    class CronJob < Base
+      private
+
+      def client
+        @deploy_group.kubernetes_cluster.client('batch/v1beta1')
```

Comment:
support 'batch/v2alpha1' for k8s <v1.8, and/or fail on older?

Reply:
`You need a working Kubernetes cluster at version >= 1.8 (for CronJob). For previous versions of cluster (< 1.8) you need to explicitly enable batch/v2alpha1 API by passing --runtime-config=batch/v2alpha1=true`
... so I'd rather use the official version ... also who would run such an old cluster :trollface:

Reply:
ok, but then CronJob should blow up / not validate on a <1.8 cluster.